sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -9,86 +9,161 @@
     /// </summary>
-    public class DefaultAzureCredential : ChainedTokenCredential
+    public class DefaultAzureCredential : TokenCredential
     {
-        private static readonly ReadOnlyMemory<TokenCredential> s_defaultCredentialChain = GetDefaultAzureCredentialChain(new DefaultAzureCredentialOptions());
+        private static readonly IExtendedTokenCredential[] s_defaultCredentialChain = GetDefaultAzureCredentialChain(CredentialPipeline.GetInstance(null), new DefaultAzureCredentialOptions());
 
+        private readonly Exception[] _unavailableExceptions;
+        private readonly IExtendedTokenCredential[] _sources;
+        private readonly CredentialPipeline _pipeline;
         /// <summary>
         /// Creates an instance of the DefaultAzureCredential class.
         /// </summary>
         /// <param name="includeInteractiveCredentials">Specifies whether credentials requiring user interaction will be included in the default authentication flow.</param>
         public DefaultAzureCredential(bool includeInteractiveCredentials = false)
-            : this(new DefaultAzureCredentialOptions { ExcludeInteractiveBrowserCredential = !includeInteractiveCredentials })
         {
+            DefaultAzureCredentialOptions options = (includeInteractiveCredentials) ? new DefaultAzureCredentialOptions { ExcludeInteractiveBrowserCredential = !includeInteractiveCredentials } : null;
 
+            _pipeline = CredentialPipeline.GetInstance(options);
+
+            _sources = GetDefaultAzureCredentialChain(_pipeline, options);
+
+            _unavailableExceptions = new Exception[_sources.Length];
         }
 
         /// <summary>
-        /// Creates an instance of the DefaultAzureCredential class.
+        /// Creates an instance of the <see cref="DefaultAzureCredential"/> class.
         /// </summary>
         /// <param name="options"></param>
         public DefaultAzureCredential(DefaultAzureCredentialOptions options)
-            : base(GetDefaultAzureCredentialChain(options))
         {
+            _pipeline = CredentialPipeline.GetInstance(options);
+
+            _sources = GetDefaultAzureCredentialChain(_pipeline, options);
+        }
+
+        /// <summary>
+        /// Sequencially calls <see cref="TokenCredential.GetToken"/> on all the specified sources, returning the first successfully retured <see cref="AccessToken"/>.
+        /// </summary>
+        /// <param name="requestContext">The details of the authentication request.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>The first <see cref="AccessToken"/> returned by the specified sources. Any credential which raises a <see cref="CredentialUnavailableException"/> will be skipped.</returns>
+        public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
```

Comment:
This method is pretty complicated, using the pattern where `bool async` as passed as a parameter might allow to reduce the duplication.

Reply:
done